diff --git a/openseadragon.module b/openseadragon.module
index 62d1ff9..39fc716 100644
--- a/openseadragon.module
+++ b/openseadragon.module
@@ -79,6 +79,11 @@ function template_preprocess_openseadragon_formatter(&$variables) {
         'id' => $openseadragon_viewer_id,
         'prefixUrl' => 'https://cdnjs.cloudflare.com/ajax/libs/openseadragon/2.4.2/images/',
         'tileSources' => $tile_sources,
+
+          // for dsu-utsc
+         'loadTilesWithAjax' => TRUE,
+         'ajaxWithCredentials' => TRUE,
+         'ajaxHeaders' => ["Authorization" => "Bearer " . \Drupal::service('jwt.authentication.jwt')->generateToken(), 'token' => \Drupal::service('jwt.authentication.jwt')->generateToken() ],
       ] + $viewer_settings,
     ];
 
@@ -124,6 +129,11 @@ function template_preprocess_openseadragon_iiif_manifest_block(&$variables) {
         'id' => $openseadragon_viewer_id,
         'prefixUrl' => 'https://cdnjs.cloudflare.com/ajax/libs/openseadragon/2.4.2/images/',
         'tileSources' => $tile_sources,
+
+        // for dsu-utsc
+        'loadTilesWithAjax' => TRUE,
+        'ajaxWithCredentials' => TRUE,
+        'ajaxHeaders' => ["Authorization" => "Bearer " . \Drupal::service('jwt.authentication.jwt')->generateToken(), 'token' => \Drupal::service('jwt.authentication.jwt')->generateToken() ],
       ] + $viewer_settings,
     ];
 
